### ISSUE TITLE
sortChanged columns no longer in priority order

### ISSUE THREAD

**Issue by rwilliams-scottlogic** [angular-ui/ui-grid#5278]
Since the change to make priority zero-based in v3.1.1, the sorting of columns by priority in `getColumnSorting` no longer works correctly. This is due to `rowSorter.priotitySort` not dealing with priority of 0 correctly in all cases:

https://github.com/angular-ui/ui-grid/commit/62dbcfe917235d827ac09755f37d0896904a99b0#diff-5158ed96a1abd5383223cd99de3d2772R361

Nothing else in the grid relies on this.

To reproduce, have a grid with columns A, B in that order. Sort B, then sort A. `getColumnSorting` should return B, A - but actually returns A, B.

Fix:

```
// Only A has a priority
else if (a.sort.priority !== undefined) {
  return -1;
}
// Only B has a priority
else if (b.sort.priority !== undefined) {
  return 1;
}
```
- [x] This is a bug report, not a question on how to use the grid.  
  Use [Stack Overflow](http://stackoverflow.com/questions/tagged/angular-ui-grid) or [Gitter](https://gitter.im/angular-ui/ui-grid) for questions.
- [x] You have searched the open issues to see if this bug has been filed before.  We don't want duplicate issues.
- [x] You have reviewed the tutorials and documentation to ensure you are using the grid correctly. [ui-grid.info](http://ui-grid.info/docs/#/tutorial)
- [x] You are using the latest version of the grid. Older versions are not patched or supported.
- [x] You have provided steps to recreate your bug.  A plunkr is even better.

**Comment by rmhubley** [comment]
I just fell into this hole as well.  Still a problem in 3.2.9

**Comment by mportuga** [comment]
This issue was fixed by the following commit on version 4.0.0:
https://github.com/angular-ui/ui-grid/commit/b6b0688c35992717963b6b4e632c7c98e3fbcaa7
